### PR TITLE
fix: remove unnecessary title attributes for label in Form

### LINKED ISSUE #46258: Form.Item <label> elements have unnecessary title attributes
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-vk3yfn?file=/index.js)

### Steps to reproduce

- Open the minimal reproduction example
- Use the browser Inspector to view the HTML for the `Form.Item` `<label>` elements
- Note that the `<label>` elements have title attributes that match the content inside the `<label>` element

### What is expected?

`<label>` elements should not use title attributes, as they are redundant and at best offer no accessibility information. At worst, in this case, they can cause confusing repeated screen reader announcements. Unless working with an `<iframe>` title is often a worse solution for accessible naming than other options (e.g. `alt`, `aria-label`, `aria-labelledby`) that are better supported across assistive technologies. In this case particularly, the content of the `<label>` does all of the work for us, so no additional attributes are needed.

References:
- [How to use title attributes accessibly](https://www.a11yproject.com/posts/title-attributes/)
- [Support of the title attribute across technologies](https://a11ysupport.io/tech/svg/title_element)

### What is actually happening?

The `<label>` element has a redundant title attribute value, which causes some assistive technologies, like screen readers, to provide an accessible name to the related `<input>` which is a duplicate value of the `<label>` content (e.g. "Name Name"). For some combinations of browsers and screen readers (e.g. VoiceOver + Chrome) the `<label>` content, while appropriately associated with the input field, won't be announced at all.

| Environment | Info |
| --- | --- |
| antd | 5.12.1 |
| React | 17.0.2 |
| System | MacOS Sonoma 14.1.2 |
| Browser | Chrome 119.0.6045.199 |

### PR DESCRIPTION
**Pull request by sunsunmonkey**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [x] Component style update
- [x] TypeScript definition update
- [x] Bundle size optimization
- [x] Performance optimization
- [x] Enhancement feature
- [x] Internationalization
- [x] Refactoring
- [x] Code style optimization
- [x] Test Case
- [x] Branch merge
- [x] Workflow
- [x] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/46258

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      remove unnecessary title attributes for label in Form     |
| 🇨🇳 Chinese |       删除表格中label不必要的标题属性    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by sunsunmonkey** [comment]
#46258

**Review by yoyo837 — APPROVED** [review]

File: components/form/FormItemLabel.tsx [hunk 1]
```diff
@@ -148,11 +148,7 @@ const FormItemLabel: React.FC<FormItemLabelProps & { required?: boolean; prefixC
 
   return (
     <Col {...mergedLabelCol} className={labelColClassName}>
-      <label
-        htmlFor={htmlFor}
-        className={labelClassName}
-        title={typeof label === 'string' ? label : ''}
-      >
+      <label htmlFor={htmlFor} className={labelClassName}>
```
Comment (afc163) [inline components/form/FormItemLabel.tsx:151]: 这样如果文字标题太长被隐藏掉后就看不见完整 label 文字了。

Comment (sunsunmonkey) [reply components/form/FormItemLabel.tsx:151]: 嘶~，确实啊，不过那个原生的也挺丑的，可以再做个hover完整提示的或许🙃

Comment (afc163) [reply components/form/FormItemLabel.tsx:151]: 原生的适配性更好，也能减少对用户的干扰。

Comment (sunsunmonkey) [reply components/form/FormItemLabel.tsx:151]: ok

**Pull request merged**